torch/jit/annotations.py

```diff
@@ -170,13 +170,14 @@ def get_type_line(source):
     type_lines = list(filter(lambda line: type_comment in line[1], lines))
     # `type: ignore` comments may be needed in JIT'ed functions for mypy, due
     # to the hack in torch/_VF.py.
-    type_lines = list(filter(lambda line: not line[1].endswith("# type: ignore"),
+
+    type_pattern = re.compile("# type: ignore(\\[.+\\])?$")
```

Comment:
I think it would be nice to add a comment explaining this regex. I had to read it over a few times and think about what was going on.

Reply:
Good point, I looked at error codes of MyPy and it seems all of them are comprised of just alphabets and hyphens, so I changed the regex accordingly. Note that the `wrong_type_pattern` is not tightened in same way because we want to catch the errors more broadly.

Reply:
> I think it would be nice to add a comment explaining this regex. I had to read it over a few times and think about what was going on.

Added comment to explain both regex.